src/resolve/resolved-dns-stub.c

```diff
@@ -460,12 +460,26 @@ static int manager_dns_stub_udp_fd(Manager *m) {
 static int manager_dns_stub_udp_fd_extra(Manager *m, DNSStubListenerExtra *l) {
         _cleanup_free_ char *pretty = NULL;
         _cleanup_close_ int fd = -1;
+        union sockaddr_union sa;
         int r;
 
         if (l->udp_fd >= 0)
                 return 0;
 
-        fd = socket(socket_address_family(&l->address), SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (l->family == AF_INET)
+                sa = (union sockaddr_union) {
+                        .in.sin_family = l->family,
+                        .in.sin_port = l->port != 0 ? htobe16(l->port) : htobe16(53),
```

Comment:
doesn't matter, but you could move the ternary op inside the htobe16()? (might need to suffix 53 with U then though)